segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/WriterState.java

```diff
@@ -43,6 +44,7 @@
     private final AtomicLong iterationId;
     private final AtomicReference<ForceFlushContext> forceFlushContext;
     private final AtomicReference<Queue<Operation>> lastRead;
+    private final AtomicInteger failedIterationCount;
```

Comment:
These member declarations scare the crap out of me!
Am I expected to believe that every single one of these is managed independently in a concurrent context and all race conditions of the combinatorics of interactions have been dealt with!? Even worse because these are mostly exposed as getters and setters it's the caller's job to do it. Honestly this whole class looks like it would have been just some local variables of some function with a loop that was crudely turned asynchronous so they were all crammed into here so they could be referenced by multiple iterations of the loop.

Reply:
We'll have to revisit this class and its whole design later. I want to keep scope of this change limited.